x-pack/plugins/infra/server/lib/adapters/metrics/lib/check_valid_node.ts

```diff
@@ -25,5 +24,7 @@ export const checkValidNode = async (
       },
     },
   };
-  return (await search(params)).hits.total > 0;
+
+  const result = await search(params);
+  return result && result.hits && result.hits.total && result.hits.total.value > 0;
```

Comment:
I've been struggling with things like this too where we need to grab a value off some deeply nested object that has optional parts all the way down 🐢. I wonder if we shouldn't just standardize on using Lodash `get` for things like this since we are only concerned with the value at the end of the path. That's was standard practice at one point in time for the Kibana team. I only started using the same pattern as you when I started with Typescript.

```javascript
const result = await search(params)
const value = get(result, 'hits.total.value', 0);
return value > 0;
```
_You don't need to make this change, unless you like it. Technically they are both the same so I will let you decide._

Reply:
The APM team has implemented some typesafer getter pattern which we might want to copy: https://github.com/elastic/kibana/blob/master/x-pack/plugins/apm/common/idx.ts

Reply:
We should settle on something because personally I don't like typing `result & result.total & result.total.value`, I like editing it even less, especially when the structure changes.

Reply:
indeed... the `idx` (which could be named better) has a few nice properties:

* preserves the return type
* auto-completion in the accessor
* simple, local implementation

Reply:
Yep, I agree it's annoying and if we can decide on a standardised solution (be it Lodash's `get`, APM's `idx`, or a tweaked custom function of our own) that'd be great. I'll merge this as is for now as it's managed to get a green build amid the CI issues. But I'll add an agenda issue for the next Infrastructure & Service Monitoring UI Sync meeting and we can decide which to use.